### PR TITLE
Building simple and interactive maps using leaflet.js

### LINKED ISSUE #3287: Building simple and interactive maps using leaflet.js
## Building simple and interactive maps using leaflet.js
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs):
Do you want to build your very own map?

Then you've come to the right place. In this tutorial, I'll show you how to build your own interactive map using leaflet.js.

Leaflet is the leading open-source JavaScript library for mobile-friendly interactive maps. Weighing just about 39 KB of JS, it has all the mapping features most developers ever need.

Leaflet is designed with *simplicity, performance* and *usability* in mind. It works efficiently across all major desktop and mobile platforms, can be extended with lots of plugins, has a beautiful, easy to use and well-documented API and a simple, readable source code that is a joy to contribute to.

### Key takeaways: 
- Integrate leaflet.js using its unique content delivery network(CDN).
- Create a simple osm map layer.
- Plot multiple markers on the map layer.

### References: 
- <a href="https://leafletjs.com/" target="_top">Leaflet js</a>

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by muyiwexy**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3287.

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Bayler** [comment]
Hi @muyiwexy Welcome to our Engineering community program. I will be your reviewer.

Before I can review your article, please follow all pre-submission checklist to the letter. You have selected almost all the check boxes yet you haven't followed the instructions. Also, all check boxes need to be followed and checked. 

Some noted mistakes:

1. Is your hero image 600px by 400px? 
2. I hope you've passed your article for a plagiarism check and made sure it's 10% or less
3. Have you checked your article for readability using the Hemmingway app?
4. Have you added sources for quotes and images that aren't yours?
5. You need an author bio. Your profile image should be named as avatar.jpg and author details saved as index.md. These two files shouldn't be put in the same folder as your article. they should be put inside this authors  [folder](https://github.com/section-engineering-education/engineering-education/tree/master/content/authors).
6. Have you used h3 and above? For example, h3, h4, h5. I see a h2 inside your article.

Etcetra.

Mistakes such as these usually warrant PR closures since these instructions have been laid out clearly, but since it's your first time submitting the PR. I'll guide you through. But avoid such mistakes in the future. 

Please make those changes then I can continue with the review.

**Comment by Bayler** [comment]
Awesome @muyiwexy All good.

Please allow me to now work on the review.

**Comment by louisefindlay23** [comment]
@muyiwexy, welcome to the EngEd programme. 

Just to let you know in the future, please fill in the `This closes #issue-number` in the Article Checklist by removing the backticks and replacing `issue-number` with the issue number of your topic suggestion so you can be correctly paid out and your topic suggestion closed when this has been merged. 

Have done this for you above so you can see the format.

**Comment by hectorkambow** [comment]
@muyiwexy 
Lets please be sure it add value beyond what is in the official docs and what is covered in other blog sites. 

The title topic seems like something a reader could easily attain from reviewing the leaflet.js documentation. 

cc @Bayler

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by muyiwexy** [comment]
@hectorkambow I used a simple topic in order to give the reader a sense of confidence that building maps are not as difficult as it seems. I also made it different from the documentation by giving a step by step explanation of some of the methods used and why they were used in which you would not find anywhere. I also gave an explanation on a possible error code the reader might experience if he fails to correctly input the DOM id of the map container.

File: content/articles/building-simple-maps-using-leaflet-js/index.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        "Marker": marker
+    }
+    L.control.layers (Basemaps, Overlaymaps).addTo (map_init);
+</script>
+</html>
+```
+If we run the above HTML code in our browser, we would have our simple, lightweight web map like the image below.
+
+![browser image](/Building-simple-and-interactive-maps-using-leaflet.js/mapimage.png)
+
```
[the remark this answers is not in the document]
Comment (muyiwexy) [reply content/articles/building-simple-maps-using-leaflet-js/index.md:229]: pardon me sir but my tutorial doesn't cover these aspects.  I can tell the reader if it is possible to do all that but showing it has taken away entirely the scope of my tutorial

**Comment by Bayler** [comment]
@muyiwexy Welcome again to our Engineering program. Your tutorial is now ready for final review. Please take note of the changes I've made on my commits so that you can better your next article.

**Comment by ahmadmardeni1** [comment]
Welcome to Section EngEd and congrats on your first article 🎉 @muyiwexy 

I tried to crop your picture but nothing worked. We only accept 600x400 pixels. Please upload another picture with the dimensions I mentioned and I will publish your article.
And make sure to go over the changes I made 37e350b

Thank you for the review @Bayler

**Comment by Bayler** [comment]
> Welcome to Section EngEd and congrats on your first article tada @muyiwexy
> 
> I tried to crop your picture but nothing worked. We only accept 600x400 pixels. Please upload another picture with the dimensions I mentioned and I will publish your article.
> And make sure to go over the changes I made [37e350b](https://github.com/section-engineering-education/engineering-education/commit/37e350b5995176b18a3af20b6c14b6f6feacf758)
> 
> Thank you for the review @Bayler

@ahmadmardeni1 which picture is to be cropped? The hero image is 600 px by 400px

**Comment by Bayler** [comment]
> I'm sorry, I didn't clarify enough. His avatar (the personal pic). @Bayler

Okay @ahmadmardeni1 Noted.

cc @muyiwexy

**Comment by ahmadmardeni1** [comment]
Take a look:

Capture

**Comment by muyiwexy** [comment]
I have updated the avatar. Thank you sir

**Comment by Bayler** [comment]
> I have updated the avatar. Thank you sir

@ahmadmardeni1 please check if it's okay now.

**Comment by ahmadmardeni1** [comment]
Great, thank you @muyiwexy 

Please make sure to double-check the code snippets and the pictures in your next reviews @Bayler. Thank you.

**Pull request merged**